sheeprl/algos/dreamer_v3/dreamer_v3.py

```diff
@@ -658,45 +658,46 @@ def main(fabric: Fabric, cfg: Dict[str, Any]):
 
         # Train the agent
         if iter_num >= learning_starts:
-            ratio_steps = policy_step - prefill_steps * policy_steps_per_iter
-            per_rank_gradient_steps = ratio(ratio_steps / world_size)
-            if per_rank_gradient_steps > 0:
-                local_data = rb.sample_tensors(
-                    cfg.algo.per_rank_batch_size,
-                    sequence_length=cfg.algo.per_rank_sequence_length,
-                    n_samples=per_rank_gradient_steps,
-                    dtype=None,
-                    device=fabric.device,
-                    from_numpy=cfg.buffer.from_numpy,
-                )
-                with timer("Time/train_time", SumMetric, sync_on_compute=cfg.metric.sync_on_compute):
-                    for i in range(per_rank_gradient_steps):
-                        if (
-                            cumulative_per_rank_gradient_steps % cfg.algo.critic.per_rank_target_network_update_freq
-                            == 0
-                        ):
-                            tau = 1 if cumulative_per_rank_gradient_steps == 0 else cfg.algo.critic.tau
-                            for cp, tcp in zip(critic.module.parameters(), target_critic.parameters()):
-                                tcp.data.copy_(tau * cp.data + (1 - tau) * tcp.data)
-                        batch = {k: v[i].float() for k, v in local_data.items()}
-                        train(
-                            fabric,
-                            world_model,
-                            actor,
-                            critic,
-                            target_critic,
-                            world_optimizer,
-                            actor_optimizer,
-                            critic_optimizer,
-                            batch,
-                            aggregator,
-                            cfg,
-                            is_continuous,
-                            actions_dim,
-                            moments,
-                        )
-                        cumulative_per_rank_gradient_steps += 1
-                    train_step += world_size
+            if cfg.algo.train_after_game or reset_envs > 0:
```

Comment:
Hi @LucaVendruscolo, I would change the name of the argument to `train_on_episode_end`.
By default its value is set to `False` and it maintains the standard behaviour.
When set to `True`, it will start training only if the episode is finished.
So, you should change the control to: `if (cfg.algo.train_on_episode_end and reset_envs > 0) or not cfg.algo.train_on_episode_end:`.

I prefer to have a configuration with a clear argument name.